File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    answer = 1
    clothes_obj = {}
    
    for item in clothes:
        [item_name, item_type] = item
        if item_type in clothes_obj:
            clothes_obj[item_type].append(item_name)
        else:
            clothes_obj[item_type] = [item_name]
    
    for item in clothes_obj:
        answer *= len(clothes_obj[item]) + 1
        
    return answer - 1

File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    
    const ref = new Map();
    clothes.forEach(([name, type]) => {
        ref.set(type, ref.has(type) ? ref.get(type) + 1 : 1);
    })
    
    const answer = [...ref.values()].reduce((sum, num) => sum * (num + 1), 1) - 1;
    
    return answer;
}